src/core/namespace.c

```diff
@@ -328,9 +328,11 @@ static int make_read_only(BindMount *m) {
 
         if (IN_SET(m->mode, INACCESSIBLE, READONLY))
                 r = bind_remount_recursive(m->path, true);
-        else if (IN_SET(m->mode, READWRITE, PRIVATE_TMP, PRIVATE_VAR_TMP, PRIVATE_DEV))
+        else if (IN_SET(m->mode, READWRITE, PRIVATE_TMP, PRIVATE_VAR_TMP, PRIVATE_DEV)) {
                 r = bind_remount_recursive(m->path, false);
-        else
+                if (r == 0 && m->mode == PRIVATE_DEV) /* can be readonly but the submounts can't*/
+                        r = mount(NULL, m->path, NULL, MS_REMOUNT|MS_NOSUID|MS_NOEXEC|MS_RDONLY, NULL);
```

Comment:
This *looks* like it could change the mount options. If I'm reading the man page right, MS_STRICTATIME cannot be changed, but just to be on the safe side, could you add
```
#define DEV_MOUNT_OPTIONS (MS_NOSUID|MS_STRICTATIME|MS_NOEXEC)
```
somewhere above and use it in `mount` and then OR it with MS_REMOUNT|MS_RDONLY here?